tests/gemm/test_specialized_gemm_routing.py

```
@@ -0,0 +1,258 @@
+    _assert_cosine(baseline, result, 0.97)
+
+
+@pytest.mark.parametrize("case_idx,item", _params("bmm_fp8", BMM_FP8_WORKLOADS))
+def test_bmm_fp8_sm121_specialized_routing(case_idx, item):
+    del case_idx
+    _skip_if_backend_missing("bmm_fp8", item["impl"])
+
+    batch = int(item["b"])
+    m, k, n = int(item["m"]), int(item["k"]), int(item["n"])
+    torch.manual_seed(batch * 10000019 + m * 1000003 + k * 101 + n)
+    input = torch.randn((batch, m, k), device="cuda", dtype=torch.bfloat16)
```

Ruff (0.15.12)

[error] Variable `input` is shadowing a Python builtin

(A001)